### ISSUE TITLE
能问下您跑的最优的模型能达到多少的reward？

### ISSUE THREAD

**Issue by codemaster17611** [Wenbing-Yao/paper-sharing#1]

**Comment by Wenbing-Yao** [comment]
dqn 还是 reinforce ? 
reinforce 平衡杆满分吧
dqn 那个打游戏的，我1080训个几小时应该最高可以达到4000+吧
<img width="546" alt="截屏2022-03-19 上午11 55 12" src="https://user-images.githubusercontent.com/52312901/159105744-0dfce326-7e4e-4aa8-a0e6-24d611e45a91.png">

**Comment by Wenbing-Yao** [comment]
我代码里用ram训练的，你可以试试用图像来做，直接用dqn论文里面的结构，再加skipframe+多张历史图片+一堆乱七八糟的tricks，应该可以训练得比较好

**Comment by Wenbing-Yao** [comment]
给模型加个dropout吧，上面图里用的下面这个模型训练的，还用了relu6：

`class ModuleInitMixin:

    def _initialize_weights(self):
        for module in self.modules():
            if isinstance(module, nn.Conv2d) or isinstance(module, nn.Linear):
                nn.init.normal_(module.weight, 0, 0.01)
                nn.init.normal_(module.bias)

class DARModelDrop(ModuleInitMixin, nn.Module):

    def __init__(self, device='cpu') -> None:
        super().__init__()
        self.fc = nn.Sequential(
            nn.Dropout(0.8),
            nn.Linear(128, 256),
            # nn.ReLU6(),
            nn.Dropout(0.5),
            nn.Linear(256, 128),
            nn.ReLU6(),
            nn.Dropout(0.5),
            nn.Linear(128, 128),
            nn.ReLU6(),
            nn.Linear(128, 64),
            nn.ReLU6(),
            nn.Linear(64, 6),
        )
        self.device = device
        self._initialize_weights()

    def forward(self, x):
        if isinstance(x, np.ndarray):
            x = torch.tensor(x).float()

        x = x.to(self.device)
        return self.fc(x)
`

**Comment by codemaster17611** [comment]
首先，非常感谢您的回复。由于我最近是一直研究图像的方式，基于cart-pole-v0的连续4帧图片，选择3层CNN提取做DQN学习只能做到80mean-score，不知道如何能提升。如果有什么好的建议可以相互交流下

**Comment by Wenbing-Yao** [comment]
RL训练，一般足够长时间训练效果应该都不错吧。
至于其它的各种tricks都用一用，比如调整一下学习率、优化器、replay buffer、loss 函数、batch size、reward reshape...
